virl2_client/models/node_image_definitions.py

```diff
@@ -82,6 +82,30 @@ def image_definitions_for_node_definition(self, definition_id: str) -> list[dict
         url = "node_definitions/" + definition_id + "/image_definitions"
         return self._session.get(url).json()
 
+    def set_image_definition_read_only(
+        self, definition_id: str, read_only: bool
+    ) -> None:
+        """
+        Set the read-only attribute of the image definition with the given ID.
+
+        :param definition_id: The ID of the image definition.
+        :param read_only: The new value of the read-only attribute.
+        """
+        url = f"image_definitions/{definition_id}/read_only"
+        self._session.put(url, json=read_only)
```

Comment:
can we change these to return a response?  I just found out that both endpoints return a full def object.
```suggestion
        return self._session.put(url, json=read_only).json()
```

Reply:
Done.